packages/astro/src/core/create-vite.ts

```diff
@@ -32,7 +32,7 @@ import { joinPaths } from './path.js';
 interface CreateViteOptions {
 	settings: AstroSettings;
 	logging: LogOptions;
-	mode: 'dev' | 'build' | string;
+	mode: string;
```

Comment:
Are we able to keep the previous type definition? It _is_ redundant for now since the `string` would widen the type, but we have discussion to circle back in refactoring `mode` in the future, and keeping `'dev' | 'build'` keeps the history here.

Reply:
Yup we can! I can see why you'd want this. I'll move https://typescript-eslint.io/rules/no-redundant-type-constituents to the _todo_ pile.

Reply:
Note that allowing redundant string literals does still improve autocomplete.

Reply:
@nickmccurdy Hm, doesn't seem to from my experience. Union-ing with a lowercase `string` will blow away other string literals in VS Code. There's a few hacks you can use to get around this like adding `& {}` or `String` to the type.